### PR TITLE
bug/SWTC-1459 missing revert data

### PR DESCRIPTION
**Pull request by JGiter**
### Summary

|             |   |
|-------------|---|
| Description | In order to send transaction to offerable identity two conditons must be met. Signer should own offerable identity and offerable identity should own `did` being updated. First is checked in OfferableIdentity.sol. This PR adds second|

Also had to pin @sphereon/pex, because recent bugfix introduced breaking changes

### CODE REVIEW

**Review by jrhender — COMMENTED** [review]
How hard would it be to add a test that covers this scenario?

File: packages/proxyIdentity/src/offerableIdentityOperator.ts [hunk 1]
```diff
@@ -77,6 +78,10 @@ export class OfferableIdenitytOperator extends Operator {
     validity?: number
   ): Promise<BigNumber> {
     const identity = addressOf(did);
+    const signer = await this.getAddress();
+    if (!((await this.identityOwner(identity)) !== signer)) {
```
Comment (nichonien) [inline packages/proxyIdentity/src/offerableIdentityOperator.ts:82]:
```suggestion
    if (!((await this.identityOwner(identity)) === signer)) {
```

**Review by nichonien — APPROVED** [review]

File: packages/proxyIdentity/src/offerableIdentityOperator.ts [hunk 2]
```diff
@@ -77,6 +83,10 @@ export class OfferableIdenitytOperator extends Operator {
     validity?: number
   ): Promise<BigNumber> {
     const identity = addressOf(did);
+    const identityOwner = await this.identityOwner(did);
+    if (!(identityOwner === this.identity.address)) {
+      throw new OnlyOwnerAllowed(identityOwner, await this.getOwnerAddress());
```
Comment (jrhender) [inline packages/proxyIdentity/src/offerableIdentityOperator.ts:88]: The error message is `Signer ${signer} is not allowed to update DID identity ${identity}`, so shouldn't `did` be the second param?
```suggestion
      throw new OnlyOwnerAllowed(identityOwner, did);
```

Comment (JGiter) [reply packages/proxyIdentity/src/offerableIdentityOperator.ts]: Do you mean that order of params in message should be the same as order in error call? I put identity on the first place in constructor, because `identity` seems to me primary in this event. So that when you write `OnlyOwnerAllowed` the first what you expect is whose owner? 
Do you think preserving order has some benefits

File: packages/credentials-interface/package.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
     "url": "git+https://github.com/energywebfoundation/ew-did-registry.git"
   },
   "dependencies": {
-    "@sphereon/pex": "^1.1.0",
+    "@sphereon/pex": "1.1.3",
```
Comment (jrhender) [inline packages/credentials-interface/package.json:50]: Is this change relevant to this PR? It's a different package, so I think it should be removed

**Review by jrhender — CHANGES_REQUESTED** [review]
Just an unrelated change to `credentials-interface` that I think should be removed

File: packages/credentials-interface/package.json (hunk 3 above)
Comment (JGiter) [reply packages/credentials-interface/package.json:50]: `@sphereon/pex` introduced breaking changes in 1.1.4

Comment (jrhender) [reply packages/credentials-interface/package.json:50]: Ok, thanks for the explanation. This maybe indeed be a good change to make. Still, I think it is separate to this PR, so would be best made in a separate PR, so that we can more easily revert/isolate the change if necessary. What do you think?

Comment (JGiter) [reply packages/credentials-interface/package.json:50]: Ok, make sence https://github.com/energywebfoundation/ew-did-registry/pull/460

**Review by jrhender — APPROVED** [review]
Thanks, looks good
